server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java

```diff
@@ -125,8 +125,12 @@ private static void compactAlignedSeries(
     List<IMeasurementSchema> measurementSchemas = new ArrayList<>();
     for (String measurement : allMeasurments) {
       // TODO: use IDTable
-      measurementSchemas.add(
-          IoTDB.metaManager.getSeriesSchema(new PartialPath(device, measurement)));
+      PartialPath p = new PartialPath(device, measurement);
+      if (!IoTDB.metaManager.isPathExist(p)) {
+        logger.info("As path {} is dropped we skip it", p.getFullPath());
+        continue;
+      }
```

Comment:
Hi, what if the deletion just takes effect between the check and the getSeriesSchema?

Reply:
That's a good catch, maybe we can just catch the exception without pre existence check?

Reply:
Yes, I prefer to just catch the exception and ignore it

Reply:
`try-catch` is added instead of pre-check.